### DISCUSSION TITLE
Besoin d'aide pour le nouveau Google consent mode v2

### DISCUSSION THREAD

**Discussion by louismaximepiton** [AmauriC/tarteaucitron.js#1199] [Q&A] [answered]
Hello,

Je suis assez nouveau dans le monde des cookies et merci de développer une alternative Open Source et relativement simple d'utilisation :pray:
J'ai une question concernant un des nouveaux paramètre de la v1.17.0:

On développe une [lib basée sur Bootstrap](https://boosted.orange.com/docs/about/cookies) et pour la documentation on utilise google tag manager (et seulement celui-ci) et on avait pour habitude d'avoir ces cookies là:
```js
tarteaucitron.init({
    hashtag: '#cookie-consent',
    cookieName: 'cookie-consent',
    orientation: 'bottom',
    bodyPosition: 'top',
    cookieslist: true,
    highPrivacy: false,
    showIcon: false,
    handleBrowserDNTRequest: true,
    useExternalCss: true,
    mandatory: false
})
tarteaucitron.user.googletagmanagerId = 'GTM-P6H78BQ';
(tarteaucitron.job = tarteaucitron.job || []).push('googletagmanager')
dataLayer = [{
    site_name: 'accessibility-boosted',
    phase: 'prod',
    page_type: 'doc',
    page_title: document.title
}]

tarteaucitron.services.googletagmanager = {
    key: 'googletagmanager',
    type: 'api',
    name: 'Google Tag Manager',
    uri: 'https://policies.google.com/privacy',
    needConsent: true,
    cookies: ['_ga', '_gat', '__utma', '__utmb', '__utmc', '__utmt', '__utmz', '__gads', '_drt_', 'FLC', 'exchange_uid', 'id', 'fc', 'rrs', 'rds', 'rv', 'uid', 'UIDR', 'UID', 'clid', 'ipinfo', 'acs'],
    js() {
        'use strict'
        if (tarteaucitron.user.googletagmanagerId === undefined) {
            return
        }

        window.dataLayer = window.dataLayer || []
        window.dataLayer.push({
            'gtm.start': new Date().getTime(),
            event: 'gtm.js'
        })
        tarteaucitron.addScript(`https://www.googletagmanager.com/gtm.js?id=${tarteaucitron.user.googletagmanagerId}`)
    }
}
```
![image](https://github.com/AmauriC/tarteaucitron.js/assets/91960143/c2d5329b-d141-4aba-ba73-a83163d51864)

Mais depuis la mise à jour en v1.17.0 on a du rajouter `googleConsentMode: false` car sinon on avait plus que le `cookies-consent`. Une idée de comment utiliser `googleConsentMode: true` ?

Désolé si la demande est vague, n'hésitez pas à demander s'il y a besoin de quelque information complémentaire.

**Comment by AmauriC** [comment, marked as answer]
Salut,
Le problème est que les cookies _ga n'ont pas été posé avant le consentement?
Si oui c'est normal, c'est le principe du consent mode.

**Reply by louismaximepiton** [reply]
Ah et bien je crois que tu as mis le doigt dessus, merci beaucoup!
